modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/group/src/release-notes/fragments/release-notes-sidetab/index.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+/* eslint-disable @liferay/portal/no-global-fetch */
+/* eslint-disable no-undef */
+
+const fetchRequest = async (input) => {
+	const response = await fetch(input, {
+		headers: {
+			'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
+			'Cache-Control': 'max-age=30, stale-while-revalidate=30',
+			'x-csrf-token': Liferay.authToken,
+		},
+	});
+
+	return response.json();
+};
+
+const restArticle = async (requestId) => {
```

Comment:
I dont have the full context here, but this seems like a lot of javascript. My first thoughts are 
1. Can we do this in a freemaker/liferay template somehow instead of needing to rewriting render logic?
2. If no, should we be using a framework such as react to handle the templating/state?
Both of these may still be no, but could be worth discussing more in order to decide the best path forward

Reply:
@ryanschuhler Yes, it is in JavaScript. Regarding your two questions, I tried to implement the initial logic with Freemarker/Liferay, but I couldn't update the request ID in Freemarker using JavaScript because Freemarker always loads first and doesn't update dynamically.
I didn't use React because I'm using Low Code and couldn't find a way to apply it to this context since it is outside the Client Extensions. When I switched to JavaScript, I used the best approach I could come up with so far, following an example that Jair used in another fragment for the Release Notes context.

Reply:
If you’d like, we can set up a meeting so I can explain why I approached it this way.

Reply:
great, I scheduled something for us to talk it through. Maybe you can show it to me working and we can brainstorm the best way to architect